File: 1379-find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree/1379-find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */

class Solution {
public:
    TreeNode* getTargetCopy(TreeNode* original, TreeNode* cloned, TreeNode* target) {
        if(original == target )
            return cloned;
        if(original->left){
            TreeNode * c= getTargetCopy(original->left, cloned ->left,target);
            if(c!= NULL){
                return c;
            }
        }
        if(original->right){
           TreeNode*c=  getTargetCopy(original->right, cloned->right, target);
            if(c!=NULL)
                return c;
        }
        return NULL;
    }
};

File: 1379-find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree/1379-find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def getTargetCopy(self, original: TreeNode, cloned: TreeNode, target: TreeNode) -> TreeNode:
        if original == target:
            return cloned
        if original.left:
            x = self.getTargetCopy(original.left, cloned.left, target)
            if x is not None:
                return x
        if original.right:
            y = self.getTargetCopy(original.right, cloned.right, target)
            if y is not None:
                return y